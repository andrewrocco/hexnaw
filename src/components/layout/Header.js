import React from 'react';
import Link from 'next/link';

export const Header = () => (
  <div>
    <Link href="/">Hex Naw</Link>
    <Link href="/about" prefetch>About</Link>
  </div>
);
