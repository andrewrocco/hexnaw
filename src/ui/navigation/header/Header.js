import React, { PureComponent } from 'react';
import { Box } from 'reflexbox/styled-components';
import classnames from 'classnames';

import { HeaderLink, HeaderLogo } from 'ui/navigation';
import { StyledHeader, StyledNav, StyledNavButton } from './Header.styles';

export class Header extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      menuOpen: false,
    };

    this.handleMenuToggle = this.handleMenuToggle.bind(this);
  }

  handleMenuToggle() {
    const { menuOpen } = this.state;
    this.setState({ menuOpen: !menuOpen });

    // toggle the body class too
    document.body.classList.toggle('menu-active');
  }

  render() {
    const { menuOpen } = this.state;
    const headerClasses = classnames(menuOpen && 'menu-open');

    return (
      <StyledHeader alignItems="center" className={headerClasses}>
        <Box mr="auto">
          <HeaderLogo />
        </Box>
        <Box>
          <StyledNav open={menuOpen}>
            <HeaderLink href="/">Hex Naw</HeaderLink>
            <HeaderLink href="/about">About</HeaderLink>
          </StyledNav>
          <StyledNavButton onClick={this.handleMenuToggle}>
            <span>{!menuOpen ? 'Open Menu' : 'Close Menu'}</span>
          </StyledNavButton>
        </Box>
      </StyledHeader>
    );
  }
}
