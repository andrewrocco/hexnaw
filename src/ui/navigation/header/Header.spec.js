import React from 'react';
import 'jest-styled-components';

import { Header } from './Header';

describe('Header Component', () => {
  context('when nothing specified', () => {
    it('should render correctly', () => {
      const subject = shallow(<Header />);
      expect(subject).toMatchSnapshot();
    });
  });
});
