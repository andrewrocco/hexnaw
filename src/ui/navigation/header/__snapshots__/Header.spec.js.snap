// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component when nothing specified should render correctly 1`] = `
.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-right: auto;
}

.c4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c6 {
  font-family: 'Gotham SSm A','Gotham SSm B',Helvetica,sans-serif;
  font-style: normal;
  font-weight: 700;
  font-size: 12px;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  color: #000000;
  display: inline-block;
  padding: 0 17px;
  -webkit-transition: color .25s ease-in-out;
  transition: color .25s ease-in-out;
}

.c6:hover {
  color: #0201FE;
}

.c2 {
  background: #000000;
  float: left;
  font-size: 20px;
  padding: 22px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3 {
  float: left;
  height: 18px;
  width: 166px;
}

.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.c5 {
  padding: 21px 22px;
}

.c7 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: transparent;
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' id='hamburger' viewBox='0 0 612 612'%3E%3Cpath d='M0 95.625v38.25h612v-38.25H0zm0 229.5h612v-38.25H0v38.25zm0 191.25h612v-38.25H0v38.25z'/%3E%3C/svg%3E");
  background-repeat: no-repeat;
  border: none;
  display: inline-block;
  height: 16px;
  position: absolute;
  top: 23px;
  right: 30px;
  width: 16px;
  z-index: 501;
}

.menu-open .c7 {
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' id='close-nav' viewBox='0 0 612 612'%3E%3Cpath fill='%23FFFFFF' d='M612 36.004L576.52.604 306 270.607 35.478.603 0 36.003l270.522 270.008L0 575.998l35.478 35.4L306 341.41l270.52 269.987 35.48-35.4L341.46 306.01'/%3E%3C/svg%3E");
}

.c7 span {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

@media only screen and (max-width:700px) {
  .c6 {
    color: #FFFFFF;
    display: block;
    padding: 17px;
  }
}

@media only screen and (max-width:700px) {
  .c0.menu-open:before {
    background: rgba(0,0,0,.95);
    border-bottom: 1px solid rgba(0,0,0,.95);
    bottom: 0;
    content: "";
    height: 100vh;
    left: 0;
    position: fixed;
    right: 0;
    top: 0;
    width: 100%;
    z-index: 500;
  }
}

@media only screen and (max-width:700px) {
  .c5 {
    box-sizing: content-box;
    left: 50%;
    opacity: 0;
    position: fixed;
    text-align: center;
    top: 50%;
    -webkit-transform: translate(-50%,-50%);
    -ms-transform: translate(-50%,-50%);
    transform: translate(-50%,-50%);
    visibility: hidden;
    width: 100%;
    z-index: 500;
  }

  .menu-open .c5 {
    opacity: 1;
    visibility: visible;
  }
}

@media only screen and (min-width:700px) {
  .c7 {
    display: none;
  }
}

<Header>
  <Styled(Styled(styled.div))
    alignItems="center"
    className=""
  >
    <div
      className="sc-AxirZ c0"
    >
      <styled.div
        mr="auto"
      >
        <div
          className="c1"
        >
          <HeaderLogo>
            <Link
              href="/"
              passHref={true}
            >
              <styled.a
                href="/"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                <a
                  className="c2"
                  href="/"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <styled.svg
                    viewBox="0 0 191 27"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <svg
                      className="c3"
                      viewBox="0 0 191 27"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        fill="#FFFFFF"
                        fillRule="evenodd"
                      >
                        <path
                          d="M21.11 9h4.505c.832 0 1.507.662 1.507 1.504v14.992c0 .83-.664 1.504-1.507 1.504h-15.02c-.832 0-1.507-.662-1.507-1.504V21h10.526c.82 0 1.496-.67 1.496-1.493V9zM.072 1.504C.072.674.735 0 1.578 0h15.02c.832 0 1.507.662 1.507 1.504v14.992c0 .83-.663 1.504-1.507 1.504H1.578c-.832 0-1.506-.662-1.506-1.504V1.504zM66.455 21.5h3.234v-5.9h5.963v5.9h3.234V6.8h-3.234v5.817H69.69V6.8h-3.235v14.7zm18.96 0h11.193v-2.877h-7.98v-3.087h6.93V12.66h-6.93V9.676h7.875V6.8H85.415v14.7zm15.957 0h3.675l3.318-5.04 3.297 5.04h3.78l-5.124-7.518 4.914-7.182h-3.675l-3.13 4.767-3.065-4.767h-3.78l4.914 7.224-5.124 7.476zm28.785 0h3.192v-9.345l7.118 9.345h2.75V6.8h-3.19v9.05l-6.89-9.05h-2.98v14.7zm18.33 0h3.297l1.344-3.297h6.216l1.344 3.297h3.38l-6.3-14.805h-2.98l-6.3 14.805zm5.796-6.153l1.953-4.767 1.953 4.767h-3.907zm18.94 6.258h2.813l3.318-9.618 3.318 9.618h2.814l5.02-14.805h-3.382l-3.045 9.954-3.298-9.996h-2.772l-3.297 9.996-3.045-9.954h-3.465l5.02 14.805z"
                        />
                      </g>
                    </svg>
                  </styled.svg>
                </a>
              </styled.a>
            </Link>
          </HeaderLogo>
        </div>
      </styled.div>
      <styled.div>
        <div
          className="c4"
        >
          <styled.nav
            open={false}
          >
            <nav
              className="c5"
              open={false}
            >
              <HeaderLink
                href="/"
              >
                <Link
                  href="/"
                  passHref={true}
                >
                  <styled.a
                    href="/"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                  >
                    <a
                      className="c6"
                      href="/"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                    >
                      Hex Naw
                    </a>
                  </styled.a>
                </Link>
              </HeaderLink>
              <HeaderLink
                href="/about"
              >
                <Link
                  href="/about"
                  passHref={true}
                >
                  <styled.a
                    href="/about"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                  >
                    <a
                      className="c6"
                      href="/about"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                    >
                      About
                    </a>
                  </styled.a>
                </Link>
              </HeaderLink>
              <HeaderLink
                href="https://thescenery.co"
              >
                <styled.a
                  href="https://thescenery.co"
                >
                  <a
                    className="c6"
                    href="https://thescenery.co"
                  >
                    The Scenery â†’
                  </a>
                </styled.a>
              </HeaderLink>
            </nav>
          </styled.nav>
          <styled.button
            onClick={[Function]}
          >
            <button
              className="c7"
              onClick={[Function]}
            >
              <span>
                Open Menu
              </span>
            </button>
          </styled.button>
        </div>
      </styled.div>
    </div>
  </Styled(Styled(styled.div))>
</Header>
`;
